pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +70,32 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+// getUnknownPaths returns a list of unknown paths.
+func (h *losslessDefaulter) getUnknownPaths(raw []byte) (sets.Set[string], error) {
+	obj := h.object.DeepCopyObject()
+	strictErrors, err := json.UnmarshalStrict(raw, obj, json.DisallowUnknownFields)
+	if err != nil {
+		return nil, err
+	}
+	paths := sets.New[string]()
+	for _, err := range strictErrors {
+		paths.Insert(err.(json.FieldError).FieldPath())
+	}
+	return paths, nil
+}
+
+// convertToFieldErrorPath converts the given path into the FieldError path.
+func convertToFieldErrorPath(path string) string {
+	// Replace /foo/bar to foo/bar
+	path, _ = strings.CutPrefix(path, "/")
+	// Replace foo/0 to foo[0]
+	re := regexp.MustCompile(`\/(\d+)$`)
+	path = re.ReplaceAllString(path, `[$1]`)
+	// Replace foo/1/bar to foo[1].bar
+	re = regexp.MustCompile(`\/(\d+)\/`)
```

Comment:
I believe this works:
```suggestion
	re :=  regexp.MustCompile(`\/(\d+)(\/|$)`)
```

Reply:
Actually, no. We need to handle two cases: `foo/1` should be replaced with `foo[1]`, and `foo/1/bar` should be replaced with `foo[1].bar`. Only in the second case we need to add a `.` at the end.